portfolio/src/main/webapp/calendar.js

```diff
@@ -69,14 +98,74 @@ function getTimeObject(timeString) {
   return new Date(currentYear, currentMonth, currentDate, timeHour, timeMinute);
 }
 
-/** Creates a single element for a new calendar event. */
+/** Creates a card element for a new calendar event. */
 function updateCalendarEventList(newCalendarEvent) {
-  const newEventElement = document.createElement('li');
-  newEventElement.innerHTML = 'New Event: ' + newCalendarEvent.name +
-      '<br>starts at ' + newCalendarEvent.startTime + '<br>ends at ' +
-      newCalendarEvent.endTime;
+  const newEventCard = document.createElement('div');
+  newEventCard.classList.add('card');
+
+  const cardBody = document.createElement('div');
+  cardBody.classList.add('card-body');
+  newEventCard.appendChild(cardBody);
+
+  const cardTitle = document.createElement('h4');
+  cardTitle.classList.add('card-title');
+  cardTitle.innerText = newCalendarEvent.name;
+  cardBody.appendChild(cardTitle);
+
+  const startTimeText = document.createElement('p');
+  startTimeText.classList.add('card-text');
+  startTimeText.innerText = newCalendarEvent.startTime;
+  cardBody.appendChild(startTimeText);
+
+  const endTimeText = document.createElement('p');
+  endTimeText.classList.add('card-text');
+  endTimeText.innerText = newCalendarEvent.endTime;
+  cardBody.appendChild(endTimeText);
+
+  const deleteButton = document.createElement('button');
+  deleteButton.classList.add('btn');
+  deleteButton.classList.add('btn-danger');
+  deleteButton.innerText = 'Delete this event';
+  cardBody.appendChild(deleteButton);
 
   const eventList = document.getElementById('new-event-list');
   eventList.innterHTML = '';
-  eventList.appendChild(newEventElement);
+  eventList.appendChild(newEventCard);
+
+  // The delete button removes the event's card from the UI.
+  deleteButton.onclick = function(newEventCard) {
+    newEventCard.target.closest('div.card').remove();
+  };
+}
+
+/**
+ * Collects and returns all the events currently displayed on the UI.
+ * @return an array of calendar events
+ */
+function collectAllEvents() {
+  // A set for all calendar events displayed on the UI.
+  // Each element in this set is a Json string.
+  const allEvents = [];
+
+  const eventList = document.getElementById('new-event-list');
+
+  // Looks at each event card and scrapes the event's name and
+  // start and end times from the HTML elements.
+  // Add all event information to a set of all Json strings.
+  eventList.childNodes.forEach((eventCard) => {
```

Comment:
Instead of using `forEach(...)` and mutating the `allEvents` array you can use `.map(...)` to transform all the elements of an array. You can use it like:
```
const allEvents = Array.from(eventList.childNodes).map((eventCard) => {
  ...
  return event;
});
```

Reply:
I see, updated.